pkg/validation/cookie.go

```diff
@@ -7,18 +7,19 @@ import (
 
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/encryption"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/util"
 )
 
-func validateCookie(o options.Cookie) []string {
+func validateCookie(o *options.Cookie) []string {
 	msgs := validateCookieSecret(o.Secret)
 
-	if o.Refresh >= o.Expire {
-		msgs = append(msgs, fmt.Sprintf(
-			"cookie_refresh (%q) must be less than cookie_expire (%q)",
-			o.Refresh.String(),
-			o.Expire.String()))
+	rv := util.RefreshValue{}
+	err := rv.Set(o.InternalRefresh)
+	if err != nil {
+		msgs = append(msgs, fmt.Sprintf("cookie_refresh: %s", err))
+	} else {
+		o.Refresh = &rv
```

Comment:
I think we are trying to purge any setup from the `validation` package move forward. @JoelSpeed might have more information where he's been moving this type of setup logic for new code.

Also, `o.Refresh` feels like an internal value. It might be best to follow this convention: https://github.com/oauth2-proxy/oauth2-proxy/blob/c7721e426ac00ae25ab9fe71cf5de6c53fa3fb0c/pkg/apis/options/options.go#L109

Though I'm not in love with that convention to be honest and that might be another area @JoelSpeed has opinions on related to his configuration refactor undertaking.

Reply:
If you would like to reorganize parsing in the future that's fine, I can't endlessly dedicate time to things that have nothing to do with the feature, sorry.

Reply:
If you leave it here and we don't get a more appropriate recommendation from @JoelSpeed, please add a TODO comment noting its inclusion so we can get it an appropriate home later.

We are actively refactoring the `validation` package to make it include no stateful changes, only configuration approval or denial.

Obscure logic mutating configuration in validation like this (https://github.com/oauth2-proxy/oauth2-proxy/blob/d67d6e3152e2a1c9614906ee9bd9890f8da9a007/pkg/validation/options.go#L76) has been the root of some major bugs and regressions from contributors lately that we've been forced to fix.

Reply:
I have fixed the conversion in light of other comments, but I will stop there.

Reply:
This conversation is not resolved, waiting for @JoelSpeed's review. Please leave it open for easy tracking.